src/rlookup.c

```diff
@@ -293,14 +293,63 @@ static RSValue *jsonValToValue(RedisModuleCtx *ctx, RedisJSON json) {
     case JSONType_Array:
     case JSONType_Object:
       japi->getJSON(json, ctx, &rstr);
-      rs_val = RS_StealRedisStringVal(rstr);
-      return rs_val;
+      return RS_StealRedisStringVal(rstr);
     case JSONType_Null:
       return RS_NullVal();
     case JSONType__EOF:
-      RS_LOG_ASSERT(0, "Cannot get here");
```

Comment:
Can you explain why you remove this assert

Reply:
It was moved after the switch statement.